src/js/stores/MesosStateStore.js

```diff
@@ -287,19 +288,24 @@ class MesosStateStore extends GetSetBaseStore {
     // Combine framework (if matching framework was found) and filtered
     // Marathon tasks. This will give you a list of framework tasks including
     // the scheduler tasks or a list of Marathon application tasks.
-    return frameworks.reduce(function(serviceTasks, framework) {
-      const { tasks = [], completed_tasks = {}, name } = framework;
+    return frameworks.reduce(function(serviceTasks, frameworkProperties) {
+      const framework = new Framework(frameworkProperties);
       // Include tasks from framework match, if service is a Framework
-      if (service instanceof Framework && name === serviceName) {
+      if (
```

Comment:
I'm super uncomfortable with all the name/id manipulation and pretty sure it should be enough to simple change `serviceName` (e6350a94a53261628755fad599c04c6e4c146b09) to  `service.getFrameworkName()` to address DCOS-17301.